TrafficCapture/dockerSolution/src/main/docker/migrationConsole/lib/console_link/console_link/api/snapshot.py

```diff
@@ -0,0 +1,128 @@
+from datetime import datetime
+from typing import Optional
+from fastapi import APIRouter, HTTPException
+from pydantic import BaseModel, field_serializer
+from console_link.models.factories import get_snapshot
+from console_link.models.snapshot import SnapshotNotStarted, SnapshotStatusUnavaliable, get_latest_snapshot_status_raw
+from console_link.api.sessions import StepState, existance_check, find_session
+import logging
+
+logging.basicConfig(format='%(asctime)s [%(levelname)s] %(message)s', level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+snapshot_router = APIRouter(
+    prefix="/snapshot",
+    tags=["snapshot"],
+)
+
+
+class SnapshotStatus(BaseModel):
+    status: StepState
+    percentage_completed: float
+    eta_ms: float | None
+    started: datetime | None = None
+    finished: datetime | None = None
+
+    class Config:
+        from_attributes = True
+
+    @field_serializer("started", "finished")
+    def serialize_completed(self, dt: datetime | None) -> str | None:
+        return dt.isoformat() if dt else None
+
+    @classmethod
+    def from_snapshot_info(cls, snapshot_info: dict) -> "SnapshotStatus":
```

Comment:
Do we have any tests for different response formats here for different versions? I know Mikayla added some python tests around the theme of spinning up different cluster versions (in a container) and testing their responses with the console link library

Reply:
Added new tests specifically for this logic